src/test/kotlin/io/github/nomisrev/service/ArticleServiceSpec.kt

```diff
@@ -0,0 +1,116 @@
+              .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+              .map { it.claimValueAsLong("id").shouldBeSome() }
+              .shouldBeRight()
+          val simonId =
+            userService
+              .register(RegisterUser(simonUsername, simonEmail, simonPw))
+              .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+              .map { it.claimValueAsLong("id").shouldBeSome() }
+              .shouldBeRight()
+          val johnId =
+            userService
+              .register(RegisterUser(johnUsername, johnEmail, johnPw))
+              .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+              .map { it.claimValueAsLong("id").shouldBeSome() }
+              .shouldBeRight()
+
+          // Create some articles
+          articleService
+            .createArticle(
+              CreateArticle(UserId(simonId), validTitle, validDescription, validBody, validTags)
+            )
+            .shouldBeRight()
+          articleService
+            .createArticle(
+              CreateArticle(UserId(johnId), validTitle, validDescription, validBody, validTags)
+            )
+            .shouldBeRight()
+
+          // Get Kaaveh's feed
+          val feed =
+            articleService
+              .getUserFeed(input = GetFeed(userId = UserId(kaavehId), limit = 20, offset = 0))
+              .shouldBeRight()
+
+          feed.articlesCount shouldBeExactly 0
+        }
+        "get kaaveh's feed when he followed simon" {
+          // Create users
+          val kaavehId =
+            userService
+              .register(RegisterUser(kaavehUsername, kaavehEmail, kaavehPw))
+              .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+              .map { it.claimValueAsLong("id").shouldBeSome() }
+              .shouldBeRight()
+          val simonId =
+            userService
+              .register(RegisterUser(simonUsername, simonEmail, simonPw))
+              .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+              .map { it.claimValueAsLong("id").shouldBeSome() }
+              .shouldBeRight()
+          val johnId =
+            userService
+              .register(RegisterUser(johnUsername, johnEmail, johnPw))
+              .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+              .map { it.claimValueAsLong("id").shouldBeSome() }
+              .shouldBeRight()
+
+          // Create some articles
+          articleService
+            .createArticle(
+              CreateArticle(UserId(simonId), validTitle, validDescription, validBody, validTags)
+            )
+            .shouldBeRight()
+          articleService
+            .createArticle(
+              CreateArticle(UserId(johnId), validTitle, validDescription, validBody, validTags)
+            )
+            .shouldBeRight()
+
+          // Do Kaaveh follow Simon
+
+          // TODO: From this point, first, issue #155 must be done.
```

Comment:
I think for completeness we need to add a couple route tests (integration tests). You can see `UserRouteSpec` and `TagRouteSpec` as examples.

I would focus on testing:
 - the validations, incorrect limit, incorrect offset and both together.
 - Articles, one empty and one with a couple of articles in the feed